Индивидуальное задание №2

poofik33 opened the pull request.

poofik33 committed:
Add Personal Task 2 in directory

add solved personal task2
add gtest

poofik33 committed:
Add script for stress test

poofik33 committed:
Add script for test

poofik33 committed:
Fix creat_test_file.py

poofik33 committed:
Update travis.yml for work with PT2

poofik33 committed:
Update README.md for PT2

poofik33 committed:
Delete unused functions

poofik33 committed:
Add input tests

added ability to input from files
added tests for input functions
added rules for travis

poofik33 committed:
Change stages to jobs in travis CI

IlyaSaneev commented on PT2/parallel_matrix.c:
```diff
@@ -0,0 +1,111 @@
+#include <wait.h>
+
+bool input_matrix(int **matrix_p, size_t *width, size_t *heigth, FILE *stream) {
+    if (!matrix_p)
```
width и heigth проверить?

IlyaSaneev reviewed (commented):
1. Добавить в ридми сравнение времени выполнения параллельного и последовательного подхода
2. Добавить цель подключающая модуль с реализациями подходов во время выполнения через dlopen

Сейчас реализовано, так что в проекте можно использовать только одну из реализаций: либо последовательную, либо параллельную. А если потребуется и то и другое то как быть?
Реорганизуй проект так, чтобы можно было воспользоваться обеими реализациями, причем желательно, чтобы в параллельном подходе в субпроцессах использовался последовательный алгоритм.

Нужно будет добавить тест, который сравнивает результаты последовательного и параллельного подхода:

{
... init data
res1 = serial_algo(data)
res2 = parallel_algo(data);
EXPECT_EQ(res1,res2);
}
в именах файлах и названиях методов стоит явным образом показать, что алгоритм  последоватлеьный или параллельный

poofik33 committed:
Redisign library with parallel algorithm

Added header for parallel algorithm
Added dlopen in main for dynamic library
Added test comparing results of serial and parallel algorithm

poofik33 committed:
Add result PT2 in README

poofik33 committed:
Update README.md

IlyaSaneev commented on PT2/test_gtest.cpp:
```diff
@@ -0,0 +1,201 @@
+    void *func = dlsym(library, "parallel_side_matrix_reflection");
+    parallel_reflection = reinterpret_cast<bool(*)(int * const, size_t, size_t)>(func);
+    if (!parallel_reflection)
+        return EXIT_FAILURE;
```
можно вынести  это в глобальное окружение тестов, вместо глобальной
го указателя на функцию parallel_reflection:
Environment* AddGlobalTestEnvironment(Environment* env);
https://github.com/google/googletest/blob/master/googletest/docs/advanced.md

IlyaSaneev commented on PT2/CMakeLists.txt:
```diff
@@ -0,0 +1,36 @@
+set(LIB_SOURCE_FILES matrix.c input.c)
+add_library(work_with_matrix STATIC ${LIB_SOURCE_FILES})
+set(PARALLEL_LIB_SOURCE_FILES parallel_matrix.c)
+add_library(parallel_work_with_matrix SHARED ${PARALLEL_LIB_SOURCE_FILES})
```
SHARED - динамически линкуется к исполняемому образу на этапе загрузки. Замени на MODULE

poofik33 committed:
Added changes for tests and cmake from review
